src/server/components/lists/searches/translators-interpreters.ts

```diff
@@ -67,115 +38,35 @@ function hasSworn(results: TranslatorInterpreterListItemGetObject[]): SwornOutpu
-    translationSpecialties: [],
+  const filterProps = {
+    countryName: validateCountryLower(country!),
+    region: decodeURIComponent(answers.region ?? ""),
+    servicesProvided: sanitiseServices(answers.services),
+    languagesProvided: sanitiseLanguages(answers.languages),
+    interpreterServices: sanitiseInterpretationTypes(answers.interpretationTypes ?? []),
+    translationSpecialties: sanitiseTranslationTypes(answers.translationTypes ?? []),
     offset: -1,
   };
 
   try {
-    const countryName = validateCountry(country);
-
-    servicesProvided = parseListValues("servicesProvided", params);
-    if (servicesProvided != null) {
-      servicesProvided = cleanTranslatorInterpreterServices(servicesProvided);
-    }
-
-    let translationSpecialties = parseListValues("translationSpecialties", params);
-    if (translationSpecialties != null) {
-      translationSpecialties = cleanTranslatorSpecialties(translationSpecialties);
-    }
-
-    let interpreterServices = parseListValues("interpreterServices", params);
-    if (interpreterServices != null) {
-      interpreterServices = cleanInterpreterServices(interpreterServices);
-    }
-
-    let languagesProvidedArray = parseListValues("languagesProvided", params);
-    if (languagesProvidedArray != null) {
-      languagesProvidedArray = cleanLanguagesProvided(languagesProvidedArray);
-    }
-
-    if (languagesProvidedArray) {
-      const cleanedLanguagesProvided = getLanguageNames(languagesProvidedArray.join(","));
-      params.languagesProvided = cleanedLanguagesProvided ?? undefined;
-
-      // populate filtered language names
-      languageNamesProvided = cleanedLanguagesProvided
-        ?.split(",")
-        .map((language: string) => {
-          return metaData.languages[language];
-        })
-        .join(", ");
-    }
-
-    if (servicesProvided) {
-      serviceNamesProvided = servicesProvided.map((service) => {
-        if (service.includes("all")) {
-          return camelCase(service);
-        }
-        const serviceName = metaData.translationInterpretationServices.find(
-          (metaDataService) => metaDataService.value.toLowerCase() === service
-        )?.value;
-        return camelCase(serviceName);
-      });
-    }
-
-    filterProps = {
-      countryName: country,
-      region,
-      servicesProvided,
-      languagesProvided: languagesProvidedArray,
-      interpreterServices,
-      translationSpecialties,
-      offset: -1,
-    };
-
-    if (countryName) {
-      allRows = await TranslatorInterpreterListItem.findPublishedTranslatorsInterpretersPerCountry(filterProps);
-    }
+    allRows = await TranslatorInterpreterListItem.findPublishedTranslatorsInterpretersPerCountry(filterProps);
   } catch (e) {
-    logger.error(`searchTranslatorsInterpreters: ${e.message}. Using these search params ${JSON.stringify(params)}`);
+    // continue with empty allRows[]
+    logger.error(`Exception searching for translators or interpreters`, e);
```

Comment:
Guess this doesn't need to have backticks anymore. I understand this is a very small change so you don't have to do it if you don't have the time.

Reply:
s'all good.. was figuring out why codeQL was complaining anyway 🙂. resolved